### PR TITLE
[[ Bug 15208 ]] Only unfocus a window when it is not the MainWindow a…

### PR DESCRIPTION
**Pull request by livecodesebastien**
…nymore, not when the KeyWindow changes

### CODE REVIEW

File: engine/src/mac-window.mm [hunk 1]
```diff
@@ -453,29 +453,42 @@ - (void)windowDidDeminiaturize:(NSNotification *)notification
...
-        m_window -> ProcessDidBecomeKey();
-    else
-    {
-        s_inside_focus_event = true;
-        m_window -> ProcessDidResignKey();
-        s_inside_focus_event = false;
-    }
+}
+
+// SN-2015-05-20: [[ Bug 15208 ]] A main window is not necessarily the key window
```
Comment (runrevmark) [inline engine/src/mac-window.mm:476]: In terms of the no-so-nice blinking cursor... Do you get any similar effects with other modal dialogs? I'm not sure there are any others (system-wise) at least which don't have a text field though.

Comment (livecodesebastien) [inline engine/src/mac-window.mm:476]: The same happens when using any of the answer or ask dialogs, have they a text field or not.

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay ca0289c

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok ca0289c670e03be1b6763aadd9a6aa9292853129

**Comment by livecode-vulcan** [comment]
😎 test success ca0289c670e03be1b6763aadd9a6aa9292853129
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/216)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/215)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/216)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/211)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/218)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/215)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/219)

**Pull request merged**